census/estimator/trainer/model.py

```diff
@@ -0,0 +1,246 @@
+  assign these (potentially) transformed features to either the DNN, or linear
+  regression portion of the model.
+
+  Args:
+    model_dir: str, the model directory used by the Classifier for checkpoints
+      summaries and exports.
+    embedding_size: int, the number of dimensions used to represent categorical
+      features when providing them as inputs to the DNN.
+    hidden_units: [int], the layer sizes of the DNN (input layer first)
+  Returns:
+    A DNNCombinedLinearClassifier
+  """
+  (gender, race, education, marital_status, relationship,
+   workclass, occupation, native_country, age,
+   education_num, capital_gain, capital_loss, hours_per_week) = INPUT_COLUMNS
+  """Build an estimator."""
+  
+  # Reused Transformations.
+  # Continuous columns can be converted to categorical via bucketization
+  age_buckets = layers.bucketized_column(
+      age, boundaries=[18, 25, 30, 35, 40, 45, 50, 55, 60, 65])
+
+  # Wide columns and deep columns.
+  wide_columns = [
+      # Interactions between different categorical features can also
+      # be added as new virtual features.
+      layers.crossed_column(
+          [education, occupation], hash_bucket_size=int(1e4)),
+      layers.crossed_column(
+          [age_buckets, race, occupation], hash_bucket_size=int(1e6)),
+      layers.crossed_column(
+          [native_country, occupation], hash_bucket_size=int(1e4)),
+      gender,
+      native_country,
+      education,
+      occupation,
+      workclass,
+      marital_status,
+      relationship,
+      age_buckets,
+  ]
+
+  deep_columns = [
+      layers.embedding_column(workclass, dimension=embedding_size),
+      layers.embedding_column(education, dimension=embedding_size),
+      layers.embedding_column(marital_status, dimension=embedding_size),
+      layers.embedding_column(gender, dimension=embedding_size),
+      layers.embedding_column(relationship, dimension=embedding_size),
+      layers.embedding_column(race, dimension=embedding_size),
+      layers.embedding_column(native_country, dimension=embedding_size),
+      layers.embedding_column(occupation, dimension=embedding_size),
+      age,
+      education_num,
+      capital_gain,
+      capital_loss,
+      hours_per_week,
+  ]
+
+  return tf.contrib.learn.DNNLinearCombinedClassifier(
+      model_dir=model_dir,
+      linear_feature_columns=wide_columns,
+      dnn_feature_columns=deep_columns,
+      dnn_hidden_units=hidden_units or [100, 70, 50, 25])
+
+
+def parse_label_column(label_string_tensor): 
+  """Parses a string tensor into the label tensor
+  Args:
+    label_string_tensor: Tensor of dtype string. Result of parsing the
+    CSV column specified by LABEL_COLUMN
+  Returns:
+    A Tensor of the same shape as label_string_tensor, should return
+    an int64 Tensor representing the label index for classification tasks, 
+    and a float32 Tensor representing the value for a regression task.
+  """
```

Comment:
Not sure if this will work for regression tasks (e.g. that it will return the same float as what is originally stored in the label column). Doesn't string_to_index_table_from_tensor return an index as opposed to the original value?

If that's the case we should remove the part of the comment saying that this will return a float32 tensor representing the value for a regression task